src/VisualStudio/Core/Test.Next/Services/SolutionServiceTests.cs

```diff
@@ -860,6 +860,51 @@ public async Task TestPartialProjectSync_ReferenceToNonExistentProject()
         var project1Checksum = await solution.CompilationState.GetChecksumAsync(project1.Id, CancellationToken.None);
     }
 
+    [Fact]
+    public async Task TestPartialProjectSync_SourceGeneratorExecutionVersion_1()
+    {
+        var code = @"class Test { void Method() { } }";
+
+        using var workspace = TestWorkspace.CreateCSharp(code);
+        using var remoteWorkspace = CreateRemoteWorkspace();
+
+        var solution = workspace.CurrentSolution;
+
+        var project1 = solution.Projects.Single();
+        var project2 = solution.AddProject("P2", "P2", LanguageNames.CSharp);
+
+        solution = project2.Solution;
+
+        var map = new Dictionary<Checksum, object>();
+        var assetProvider = new AssetProvider(
+            Checksum.Create(ImmutableArray.CreateRange(Guid.NewGuid().ToByteArray())), new SolutionAssetCache(), new SimpleAssetSource(workspace.Services.GetService<ISerializerService>(), map), remoteWorkspace.Services.GetService<ISerializerService>());
+
+        // Do the initial full sync
+        await solution.AppendAssetMapAsync(map, CancellationToken.None);
+        var solutionChecksum = await solution.CompilationState.GetChecksumAsync(CancellationToken.None);
+        var fullSyncedSolution = await remoteWorkspace.GetTestAccessor().GetSolutionAsync(assetProvider, solutionChecksum, updatePrimaryBranch: true, CancellationToken.None);
+        Assert.Equal(2, fullSyncedSolution.Projects.Count());
+
+        // Update the source generator versions for all projects for the local workspace.
+        workspace.EnqueueUpdateSourceGeneratorVersion(projectId: null, forceRegeneration: true);
+        await GetWorkspaceWaiter(workspace).ExpeditedWaitAsync();
+        solution = workspace.CurrentSolution;
+
+        // Now just sync project1's cone over.  This will validate that that we get the right checksums, even with a
+        // partial cone sync.
+        {
+            await solution.AppendAssetMapAsync(map, project1.Id, CancellationToken.None);
+            var project1Checksum = await solution.CompilationState.GetChecksumAsync(project1.Id, CancellationToken.None);
+            var project1SyncedSolution = await remoteWorkspace.GetTestAccessor().GetSolutionAsync(assetProvider, project1Checksum, updatePrimaryBranch: false, CancellationToken.None);
```

Comment:
Did this need an extra assert?

Reply:
No.  The impl actually already asserts that the checksums are the same (and throws if not).  (so this failed prior to this fix).